archive/100-theorems-list/9_area_of_a_circle.lean

```diff
@@ -0,0 +1,115 @@
+  disc r = region_between (λ x, -sqrt (r^2 - x^2)) (λ x, sqrt (r^2 - x^2)) (Ioc (-r) r) :=
+begin
+  ext p,
+  simp only [disc, region_between, mem_set_of_eq, mem_Ioo, mem_Ioc, pi.neg_apply],
+  split;
+  intro h,
+  { cases abs_lt_of_sqr_lt_sqr' (lt_of_add_lt_of_nonneg_left h (pow_two_nonneg p.2)) r.2,
+    rw [add_comm, ← lt_sub_iff_add_lt] at h,
+    exact ⟨⟨left, right.le⟩, sqr_lt.mp h⟩ },
+  { rw [add_comm, ← lt_sub_iff_add_lt],
+    exact sqr_lt.mpr h.2 },
+end
+
+/-- The disc is a `measurable_set`. -/
+theorem measurable_set_disc : measurable_set (disc r) :=
+by apply measurable_set_lt; apply continuous.measurable; continuity
+
+/-- The area of a disc with radius `r` is `π * r ^ 2`. -/
+theorem area_disc : volume (disc r) = nnreal.pi * r ^ 2 :=
+begin
+  let f := λ x, sqrt (r ^ 2 - x ^ 2),
+  let F := λ x, (r:ℝ) ^ 2 * arcsin (r⁻¹ * x) + x * sqrt (r ^ 2 - x ^ 2),
+  have hf : continuous f := by continuity,
+  suffices : ∫ x in -r..r, 2 * f x = nnreal.pi * r ^ 2,
+  { have h : integrable_on f (Ioc (-r) r) :=
+      (hf.integrable_on_compact compact_Icc).mono_set Ioc_subset_Icc_self,
+    calc  volume (disc r)
+        = volume (region_between (λ x, -f x) f (Ioc (-r) r)) : by rw disc_eq_region_between
+    ... = ennreal.of_real (∫ x in Ioc (-r:ℝ) r, (f - has_neg.neg ∘ f) x) :
+          volume_region_between_eq_integral
+            h.neg h measurable_set_Ioc (λ x hx, neg_le_self (sqrt_nonneg _))
+    ... = ennreal.of_real (∫ x in (-r:ℝ)..r, 2 * f x) : by simp [two_mul, integral_of_le]
+    ... = nnreal.pi * r ^ 2 : by rw_mod_cast [this, ← ennreal.coe_nnreal_eq], },
+  obtain ⟨hle, (heq | hlt)⟩ := ⟨nnreal.coe_nonneg r, hle.eq_or_lt⟩, { simp [← heq] },
+  have hderiv : ∀ x ∈ Ioo (-r:ℝ) r, has_deriv_at F (2 * f x) x,
+  { rintros x ⟨hx1, hx2⟩,
+    convert ((has_deriv_at_const x ((r:ℝ)^2)).mul ((has_deriv_at_arcsin _ _).comp x
+      ((has_deriv_at_const x (r:ℝ)⁻¹).mul (has_deriv_at_id' x)))).add
+        ((has_deriv_at_id' x).mul (((has_deriv_at_id' x).pow.const_sub ((r:ℝ)^2)).sqrt _)),
+    { have h : sqrt (1 - x ^ 2 / r ^ 2) * r = sqrt (r ^ 2 - x ^ 2),
+      { rw [← sqrt_sqr hle, ← sqrt_mul, sub_mul, sqrt_sqr hle, div_mul_eq_mul_div_comm,
+            div_self (pow_ne_zero 2 hlt.ne'), one_mul, mul_one],
+        simpa [sqrt_sqr hle, div_le_one (pow_pos hlt 2)] using sqr_le_sqr' hx1.le hx2.le },
+      field_simp,
+      rw [h, mul_left_comm, ← pow_two, neg_mul_eq_mul_neg, mul_div_mul_left (-x^2) _ two_ne_zero,
+          add_left_comm, div_add_div_same, tactic.ring.add_neg_eq_sub, div_sqrt, two_mul] },
+    { suffices : -(1:ℝ) < r⁻¹ * x, by exact this.ne',
+      calc -(1:ℝ) = r⁻¹ * -r : by simp [hlt.ne']
+              ... < r⁻¹ * x : by nlinarith [inv_pos.mpr hlt] },
+    { suffices : (r:ℝ)⁻¹ * x < 1, by exact this.ne,
+      calc (r:ℝ)⁻¹ * x < r⁻¹ * r : by nlinarith [inv_pos.mpr hlt]
+                   ... = 1 : inv_mul_cancel hlt.ne' },
+    { nlinarith } },
+  have hcont := (by continuity : continuous F).continuous_on,
```

Comment:
```suggestion
  have hcont : continuous_on F _ := (by continuity).continuous_on,
```
Does this work?  If it does, I think it's more readable to move the type annotation outwards like this.

Reply:
Unfortunately not.